### PR TITLE
Paraview: support Windows

### PR DESCRIPTION
**Pull request by johnwparent**
Prevent use of x11
Don't define mpi compilers in cmake interface as MSMPI has no compiler wrapper.

### CODE REVIEW

File: var/spack/repos/builtin/packages/paraview/package.py [hunk 1]
```diff
@@ -385,10 +386,11 @@ def nvariant_bool(feature):
 
         rendering = variant_bool("+opengl2", "OpenGL2", "OpenGL")
         includes = variant_bool("+development_files")
+        use_x11 = nvariant_bool("+osmesa") if not sys.platform == "win32" else "OFF"
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/paraview/package.py]: We should conditionally define the `+osmesa` variant if it should always be off on Windows.

File: var/spack/repos/builtin/packages/paraview/package.py [hunk 2]
```diff
@@ -492,15 +494,19 @@ def nvariant_bool(feature):
...
+                "-DMPIEXEC:FILEPATH=%s/bin/mpiexec" % spec["mpi"].prefix,
+            ]
+            if not sys.platform == "win32":
+                mpi_args.extend(
+                    [
+                        "-DMPI_CXX_COMPILER:PATH=%s" % spec["mpi"].mpicxx,
+                        "-DMPI_C_COMPILER:PATH=%s" % spec["mpi"].mpicc,
+                        "-DMPI_Fortran_COMPILER:PATH=%s" % spec["mpi"].mpifc,
+                    ]
+                )
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/paraview/package.py:518]: (no change required) this reminds me of several other common patterns for cmake arguments that could be pulled into the CMake package itself (ROCm support is another example).

**Review by scheibelp — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/paraview/package.py (hunk 2 above)
Comment (johnwparent) [reply var/spack/repos/builtin/packages/paraview/package.py:518]: I'll make a note and next time I'm modifying the CMake I'll see if I can refactor this.

File: var/spack/repos/builtin/packages/paraview/package.py (hunk 1 above)
Comment (johnwparent) [reply var/spack/repos/builtin/packages/paraview/package.py]: I'm actually unsure if mesa is supported on Windows, perhaps it should be skipped, but I'm not super well versed on graphics libraries and their Windows support. However, whether we want to use mesa or not on Windows is not the issue here, it's that `x11` should never be used on Windows rather that being conditional on `osmesa` as it is on *nix platforms.

Comment (scheibelp) [reply var/spack/repos/builtin/packages/paraview/package.py]: OK that makes sense. It looks like `osmesa` can in fact be used on Windows.

My only request here then is to use `spec.satisfies(platform=windows)` (vs. `sys.platform`).

**Review by scheibelp — APPROVED** [review]

**Comment by johnwparent** [comment]
Pulling develop in in the hopes of avoiding the codecov failure

**Pull request merged**